jobs/keepalived/templates/bin/action.sh.erb

```diff
@@ -32,5 +37,24 @@ if [[ "$iaas_name" == "aws" ]]; then
     sudo ip addr del $virtual_ip/32 dev eth0
     set -e
   fi
+elif [[ "$iaas_name" == "openstack"  ]]; then
+  # shellcheck disable=SC1091
+  source /var/vcap/jobs/keepalived/bin/utils/openstack/generate_token
+  source /var/vcap/jobs/keepalived/bin/utils/openstack/port_operations
+  this_port_id=$(get_port_id "${SOURCE_IP}")
+  if [[ "$STATE" == "MASTER" ]]; then
+    port_update_response=$(update_port "${this_port_id}")
+    peer_port_id=$(get_port_id "${PEER_IP}")
+    port_update_response=$(update_port "${peer_port_id}")
+    port_update_response=$(update_port "${this_port_id}" "${virtual_ip}")
+    if [[ "$(sudo ip addr show | grep "${virtual_ip}")" != *"${virtual_ip}"* ]]; then
```

Comment:
this code is also same as line 32-34, can we move it to a separate function?

Reply:
one more thing, if at all we don't move this code for some reason, can we make both the code to look identical? There are a bit of differences in coding styles in both pieces here. The change should affect functionality in any way.

Reply:
done